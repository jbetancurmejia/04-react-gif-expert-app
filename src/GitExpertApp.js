import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GitExpertApp = ({ defaultCategories=[]}) => {
  //const categories = ["One punch", "Samuari X", "Dragon Ball"];
  // const [categories, setCategories] = useState(["One punch"]);
  const [categories, setCategories] = useState( defaultCategories );

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />

      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (            
          <GifGrid 
          key={category}
          category={category} />
        ))}
      </ol>
    </>
  );
};

export default GitExpertApp;
