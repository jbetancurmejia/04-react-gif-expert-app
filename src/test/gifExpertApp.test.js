import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GitExpertApp from '../GitExpertApp';


describe('Pruebas en componente <GitExpertApp/>', () => {
  
    test('debe mostrarse correctamente', () => {

        const wrapper = shallow( <GitExpertApp/> );
      expect( wrapper ).toMatchSnapshot();
    });


    test('debe de mostrar una lista de categorias', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GitExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

        
    })

    
    
});
