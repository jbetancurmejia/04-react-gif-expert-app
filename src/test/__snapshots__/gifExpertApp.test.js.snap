// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GitExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en componente <GitExpertApp/> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
